sycl/doc/EnvironmentVariables.md

```diff
@@ -28,6 +29,56 @@ compiler and runtime.
+
+
+Possible values of `backend` are:
+- `level_zero`
+- `opencl`
+- `cuda`
+- `hip`
+- `esimd_emulator`
+- `*`
+
+Possible named device types are:
+- `cpu`
+- `gpu`
+- `acc`
+- `*`
+
+To see the exact numeric device id of each of the available devices, simply run `sycl-ls`.
+
+Example: `ONEAPI_DEVICE_SELECTOR=opencl:2.0` This would select the first tile of some device with an id of 2 from OpenCL platform. 
+
+Common Usage Examples:
+
+| Example | Result |
+-----------|---------
+| `ONEAPI_DEVICE_SELECTOR=opencl:*` | All the devices on the OpenCL platform are available |
+| `ONEAPI_DEVICE_SELECTOR=level_zero:gpu` | Only gpu devices on the LevelZero platform will be available.|
+| `ONEAPI_DEVICE_SELECTOR="opencl:gpu;level_zero:gpu"` | GPU devices from both Level Zero and OpenCL will be available. (Note that escaping (like quotation marks) will likely be needed when using semi-colon separated entries.) |
+| `ONEAPI_DEVICE_SELECTOR=opencl:gpu,cpu` | Only cpu and gpu devices on the OpenCL platform will be available.|
+| `ONEAPI_DEVICE_SELECTOR=opencl:0` | Device with numeric identifier of 0 chosen from the OpenCL platform. |
+| `ONEAPI_DEVICE_SELECTOR=hip:0,2` | Devices with numeric identifier of 0 and 2 are chosen from the HIP platform. |
+| `ONEAPI_DEVICE_SELECTOR=opencl:0.*` | All the sub-devices (aka tiles) from the OpenCL device identified by 0 will be available. |
+| `ONEAPI_DEVICE_SELECTOR=opencl:0.2` | The third tile (2 in 0 based counting) of the OpenCL device with numeric identifier 0 will be the sole device available.  |
+| `ONEAPI_DEVICE_SELECTOR=level_zero:gpu,gpu.*,gpu.*.*` | A wide net is cast. All level_zero gpus will be availabe in the device list, along with all their child partitions (if any), as well as the next level of partitions below them.  It would be up to the developer to figure out which device is which, if that is important. (This can be done with `info::device::parent_device`)|
+
+
+Notes:
+- The backend argument is always required. An error will be thrown if it is absent.
+- Additionally, the backend MUST be followed by colon ( : ) and at least one device specifier of some sort, else an error is thrown.
+- For sub-devices/tiles, the parent device must support partitioning ( see`info::partition_by_domain` and `info::next_partitionable` in the SYCL 2020 Spec)
+- if using semi-colons ( ; ) to separate entries, quotation marks will likely be needed around the entire environment variable. 
```

Comment:
I think it would be better to say something like:

> The semi-colon character (`;`) is treated specially by many shells, so you may need to enclose the string in quotes if the selection string contains this character.